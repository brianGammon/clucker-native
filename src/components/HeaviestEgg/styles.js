import { StyleSheet } from 'react-native';

const row = {
  padding: 4,
  flexDirection: 'row',
};

const label = {
  fontWeight: 'bold',
  color: 'grey',
};

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  statsLabel: {
    fontWeight: 'bold',
    fontSize: 17,
  },
  row,
  headerRow: {
    ...row,
    backgroundColor: '#eee',
  },
  medLabel: {
    ...label,
    width: 100,
  },
  wideLabel: {
    ...label,
    flex: 1,
  },
  value: {
    width: 100,
  },
  wideValue: {
    flex: 1,
  },
});
