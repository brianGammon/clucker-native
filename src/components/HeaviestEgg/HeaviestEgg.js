/* @flow */
import React from 'react';
import moment from 'moment';
import { View, Text } from 'native-base';
import styles from './styles';
import { type Egg } from '../../types';

type Props = {
  heaviest: Egg,
};

const HeaviestEgg = ({ heaviest }: Props) => (
  <View style={styles.container}>
    <Text style={styles.statsLabel}>Heaviest</Text>
    <View style={[styles.row, styles.headerRow]}>
      <Text style={styles.wideLabel}>Chicken</Text>
      <Text style={styles.medLabel}>Weight(g)</Text>
      <Text style={styles.medLabel}>Date</Text>
    </View>
    <View style={styles.row}>
      <Text style={styles.wideValue}>{heaviest.chickenName}</Text>
      <Text style={styles.value}>{heaviest.weight}</Text>
      <Text style={styles.value}>
        {moment(heaviest.date).format('MMM D, YYYY')}
      </Text>
    </View>
  </View>
);

export default HeaviestEgg;
