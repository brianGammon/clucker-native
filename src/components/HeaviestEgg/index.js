import HeaviestEgg from './HeaviestEgg';

export default HeaviestEgg;
