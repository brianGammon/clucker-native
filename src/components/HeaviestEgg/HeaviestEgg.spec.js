import React from 'react';
import HeaviestEgg from '.';

describe('HeaviestEgg', () => {
  const heaviest = {
    chickenName: 'Bossie',
    chickenId: 'chicken1',
    weight: '66.6',
    date: '2018-08-01',
  };
  test('renders', () => {
    const wrapper = shallow(<HeaviestEgg heaviest={heaviest} />);
    expect(wrapper).toMatchSnapshot();
  });
});
