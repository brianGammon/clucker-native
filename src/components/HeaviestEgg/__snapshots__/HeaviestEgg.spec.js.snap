// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaviestEgg renders 1`] = `
<Styled(ViewNB)
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <Styled(Text)
    style={
      Object {
        "fontSize": 17,
        "fontWeight": "bold",
      }
    }
  >
    Heaviest
  </Styled(Text)>
  <Styled(ViewNB)
    style={
      Array [
        Object {
          "flexDirection": "row",
          "padding": 4,
        },
        Object {
          "backgroundColor": "#eee",
          "flexDirection": "row",
          "padding": 4,
        },
      ]
    }
  >
    <Styled(Text)
      style={
        Object {
          "color": "grey",
          "flex": 1,
          "fontWeight": "bold",
        }
      }
    >
      Chicken
    </Styled(Text)>
    <Styled(Text)
      style={
        Object {
          "color": "grey",
          "fontWeight": "bold",
          "width": 100,
        }
      }
    >
      Weight(g)
    </Styled(Text)>
    <Styled(Text)
      style={
        Object {
          "color": "grey",
          "fontWeight": "bold",
          "width": 100,
        }
      }
    >
      Date
    </Styled(Text)>
  </Styled(ViewNB)>
  <Styled(ViewNB)
    style={
      Object {
        "flexDirection": "row",
        "padding": 4,
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "flex": 1,
        }
      }
    >
      Bossie
    </Styled(Text)>
    <Styled(Text)
      style={
        Object {
          "width": 100,
        }
      }
    >
      66.6
    </Styled(Text)>
    <Styled(Text)
      style={
        Object {
          "width": 100,
        }
      }
    >
      Aug 1, 2018
    </Styled(Text)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;
